### DISCUSSION TITLE
[Feature]: Support file formats avif

### DISCUSSION THREAD

**Discussion by basings** [immich-app/immich#1686] [Feature Request]
### Feature detail

It would be really cool if immich would support avif. This is probably a feature request for the distant future but I wanted to open it anyway to draw attention to it.

This feature request initially intended to support uploading, processing and viewing avif files but there is also a second use case for avif. Namely to use it for the thumbnails since avif has a higher quality at the same size or a lower size with the same quality compared to webp. But maybe it's too early if the intend is to cover a very broad audience and avif is not yet supported on all devices. 

https://aomedia.org/avif/
https://aomediacodec.github.io/av1-avif/

Platform: all platforms

### Platform

Web

**Comment by gianni-rosato** [comment]
I'll second this by asking for JXL support as well. JXL can losslessly transcode JPEG images with a ~20% file size reduction, so enabling it by default server side would save users storage & bandwidth effortlessly

**Reply by basings** [reply]
jxl is not supported anymore by chromium. Probably due to the influence of Google on avif (or lack of influence on jxl). Afaik there's no official statement WHY they removed it from chromium. It'll take years until it comes back, if ever, and more years until it's widely supported and accepted...

**Reply by gianni-rosato** [reply]
There's a universally available patch for chromium that works flawlessly, and could be integrated into a native app. or images could be transcoded again. Even just seeing support for uploading JXLs would be nice

**Reply by zackpollard** [reply]
We won't be looking into using JXL or AVIF for thumbnails due to the lack of support across devices. We are however looking into optimising further the way that thumbnails work in the future to reduce the amount of data sent to the client based on screen resolution.

Support for AVIF in general though makes sense and a PR to add support would be great, otherwise we will get around to it in future 😄

**Reply by uhthomas** [reply]
JPEG XL issue: #2743

**Reply by uhthomas** [reply]
I understand this request is for thumbnails, but a good start is supporting the media type at all :)

**Comment by awoimbee** [comment]
I just got hit by `ERROR [AssetUploadConfig] Unsupported file type .avif file MIME type image/avif`. I started using AVIF to upload high quality pictures to gPhotos without using too much storage.
Immich already supports AV1 videos. It seems like every modern browser/phone supports AVIF, so I guess the only change required is to add the format to a whitelist ? [This line](https://github.com/immich-app/immich/blob/0d543bbb0a7d5ee0ea28be003df85764eda8623c/server/apps/immich/src/config/asset-upload.config.ts#L57) ?

EDIT: indeed, I changed this line (in the generated javascript) and everything is working fine (EXIF, machine learning, ...).

**Comment by elliotclee** [comment]
Hi,

I tried making a patch to support .avif upload. @awoimbee gave me a clue where to start, but there was other stuff to add. I also added support for Canon RAW (.CR2 - image/x-canon-cr2) while I was at it, since I have a number of those in my archive.

Anyways, https://github.com/immich-app/immich/compare/main...elliotclee:immich:avif-and-more?expand=1 has the patch I came up with. I can turn it into a pull request if desired.

**Comment by uhthomas** [comment]
Resolved by #2834.
